api/turing/cluster/knative_service.go

```diff
@@ -23,8 +23,9 @@ var knativeSvcDefaults = struct {
 	// to requests
 	RequestTimeoutSeconds int
 }{
-	AutoscalingClass:      "kpa.autoscaling.knative.dev",
-	RequestTimeoutSeconds: 30,
+	AutoscalingClass: "kpa.autoscaling.knative.dev",
+	// TODO: was 30s
+	RequestTimeoutSeconds: 600,
```

Comment:
As opposed to doing this, can we increase the configurable value [here](https://github.com/gojek/turing/blob/main/api/turing/config/config.go#L206) which controls the overall deployment timeout? Then, we can simply increase it as a config override for [e2e tests](https://github.com/gojek/turing/blob/main/infra/e2e/turing.values.yaml).

Reply:
That being said, increasing the timeout has the potential to make the e2e tests themselves slow (where we are testing a bad config). So I'm wary of increasing this permanently unless there is a strong need.

Reply:
Yes, let me do some more investigation before we determine this number. But I'll pull this into a configuration.

Reply:
> But I'll pull this into a configuration.

Cool. Just to clarify - if we are intending to tweak  the Knative `RequestTimeoutSeconds` here, we may need to make a new config for this. Also, in this case, it would be important to determine and note down why this needs to be modified to begin with (i.e., why tweaking the overall deployment timeout is not sufficient).

If the existing Deployment timeout config can achieve this, that's preferable.

Reply:
The rest of the changes in this PR LGTM.